### PR TITLE
[JUJU-1189] Fix failing charm storage test

### PR DESCRIPTION
**Pull request by basebandit**
This PR fixes failing test-charm-storage in jenkins. It adds checks for existence of a storage unit before asserting its properties.

## Checklist

- ~[ ] Code style: imports ordered, good names, simple structure, etc~
- ~[ ] Comments saying why design decisions were made~
- ~[ ] Go unit tests, with comments saying what you're testing~
- [x] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
 cd tests
./main.sh -v -p aws -c aws storage run_charm_storage
```

### CODE REVIEW

**Review by jack-w-shaw — CHANGES_REQUESTED** [review]
QA if failing for me. After the charm_storage test passes, and the model is destroyed, it gets stuck with status:
```
$ juju status
Model               Controller     Cloud/Region   Version   SLA          Timestamp       Notes
test-charm-storage  ctrl-32ha23tm  aws/us-east-1  2.9.35.1  unsupported  13:53:21+01:00  attempt 4 to destroy model failed (will retry):  model not empty, found 1 machine, 1 application (model not empty)

App               Version  Status   Scale  Charm             Channel  Rev  Exposed  Message
dummy-storage-mp           waiting    0/1  dummy-storage-fs             3  no       waiting for machine

Unit                Workload  Agent       Machine  Public address  Ports  Message
dummy-storage-mp/0  waiting   allocating  4                               waiting for machine

Machine  State  Address  Inst id  Series  AZ  Message
4        down            pending  jammy       cannot record provisioning info for "i-07fb514d310e8a4e5": cannot set instance data for machine "4": machine is not found or not alive
```
Logs from the pipeline also show:
```
Filesystem PASSED
Located local charm "dummy-storage-fs", revision 3
Deploying "dummy-storage-mp" from local charm "dummy-storage-fs", revision 3 on jammy
ERROR getting details for storage data/5: filesystem for storage instance "data/5" not found
removing application dummy-storage-fs
- will remove storage data/0
removing application dummy-storage-lp
- will remove storage disks/1
- will remove storage disks/2
removing application dummy-storage-tp
- will remove storage data/3
removing application dummy-storage-np
- will remove storage data/4
removing application dummy-storage-mp
- will remove storage data/5
All charm storage tests PASSED
====> Destroying juju model test-charm-storage
```

File: testcharms/charms/dummy-storage-fs/hooks/install [hunk 1]
```diff
@@ -1,16 +1,2 @@
...
-juju-log -l INFO "Data on drive: $(cat /srv/single-fs/test.txt)"
-
-for unit in $(storage-list); do
-    juju-log -l INFO "Storage details for ${unit}: $(storage-get -s $unit)"
-done
-
-# The storage-attached hook has fired so we can write some data there now.
-status-set maintenance "Installed" || true
+#!/bin/sh
+echo install
```
Comment (anvial) [inline testcharms/charms/dummy-storage-fs/hooks/install:2]: need to add a newline at the end of the file.

File: testcharms/charms/dummy-storage-lp/hooks/install [hunk 2]
```diff
@@ -1,16 +1,2 @@
...
-juju-log -l INFO "Data on drive: $(cat /srv/data/test.txt)"
-
-for unit in $(storage-list); do
-    juju-log -l INFO "Storage details for ${unit}: $(storage-get -s $unit)"
-done
-
-# The storage-attached hook has fired so we can write some data there now.
-status-set maintenance "Installed" || true
+#!/bin/sh
+echo install
```
Comment (anvial) [inline testcharms/charms/dummy-storage-lp/hooks/install:2]: need to add a newline at the end of the file.

File: testcharms/charms/dummy-storage-np/hooks/install [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+echo install
```
Comment (anvial) [inline testcharms/charms/dummy-storage-np/hooks/install:2]: need to add a newline at the end of the file.

File: testcharms/charms/dummy-storage-tp/hooks/install [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+echo install
```
Comment (anvial) [inline testcharms/charms/dummy-storage-tp/hooks/install:2]: same

**Review by anvial — CHANGES_REQUESTED** [review]
Minor changes

**Review by anvial — APPROVED** [review]
QA steps passed. 

lg2m

**Pull request merged**